operator/pkg/client/users/client.go

```diff
@@ -173,11 +176,19 @@ func (c *Client) getPassword(ctx context.Context, user *redpandav1alpha2.User) (
 				return "", err
 			}
 
+			if ptr.Deref(auth.Password.NoGenerate, false) {
+				return "", errors.Newf("Kubernetes Secret not found %s/%s", user.Namespace, secret)
+			}
+
 			return c.generateAndStorePassword(ctx, user, userProvidedPassword, nn, key)
 		}
 
 		data, ok := passwordSecret.Data[key]
 		if !ok {
+			if ptr.Deref(auth.Password.NoGenerate, false) {
+				return "", errors.Newf("Key %s not found in Kubernetes Secret %s/%s", key, user.Namespace, secret)
```

Comment:
nit(s):
- golang convention is to have errors start with lowercase letters. That way when you use `error.Wrap` there's no mixed casing.
- `%q` is preferable when logging unformatted user supplied data as it will clearly distinguish between value and error message.
- `Kubernetes` is redundant in this context
```suggestion
				return "", errors.Newf("key %q not found in Secret %s/%s", key, user.Namespace, secret)
```

Reply:
Thanks changed as suggested.